pkg/subctl/cmd/join.go

```diff
@@ -221,6 +234,32 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	exitOnError("Error deploying Submariner", err)
 }
 
+func checkRequirements(config *rest.Config) (*[]string, error) {
+	failedRequirements := []string{}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return &failedRequirements, errors.WithMessage(err, "error creating API server client")
+	}
+	serverVersion, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return &failedRequirements, errors.WithMessage(err, "error obtaining API server version")
+	}
+	major, err := strconv.Atoi(serverVersion.Major)
+	if err != nil {
+		return &failedRequirements, errors.WithMessagef(err, "error parsing API server major version %v", serverVersion.Major)
+	}
+	minor, err := strconv.Atoi(serverVersion.Minor)
+	if err != nil {
+		return &failedRequirements, errors.WithMessagef(err, "error parsing API server minor version %v", serverVersion.Minor)
+	}
+	if major < minK8sMajor || (major == minK8sMajor && minor < minK8sMinor) {
+		failedRequirements = append(failedRequirements,
+			fmt.Sprintf("Submariner requires Kubernetes %d.%d; your cluster is running %s.%s",
+				minK8sMajor, minK8sMinor, serverVersion.Major, serverVersion.Minor))
+	}
+	return &failedRequirements, nil
```

Comment:
Why not return `[]string` for failed requirements and nil for success, ie

     
```
        clientset, err := kubernetes.NewForConfig(config)
	if err != nil {
		return nil, errors.WithMessage(err, "error creating API server client")
	}
	 ...
        var failedRequirements []string
	if major < minK8sMajor || (major == minK8sMajor && minor < minK8sMinor) {
		failedRequirements = append(failedRequirements,
			fmt.Sprintf("Submariner requires Kubernetes %d.%d; your cluster is running %s.%s",
				minK8sMajor, minK8sMinor, serverVersion.Major, serverVersion.Minor))
	}
	return failedRequirements, nil
```

Reply:
I wanted to start a pattern where, if additional requirements are added in the future, if an error occurs we can still show the missing requirements we’ve identified so far (which sounds weird from a Java background...).

Reply:
Sorry I'm not sure I follow, ie I'm not seeing the reason for returning `*[]string` instead of `[]string`.  Either way it's not a big deal.